pkg/rc/replication_controller.go

```diff
@@ -785,8 +786,31 @@ func (rc *replicationController) updateAllocations(ineligible []types.NodeName)
 		},
 	}
 
+	rc.mu.Lock()
```

Comment:
I just remembered I never reviewed your rc watch pr. I'll get on that. Sorry about not doing it sooner!